sdk/src/simple_vote_transaction_checker.rs

```diff
@@ -0,0 +1,14 @@
+use {crate::pubkey::Pubkey, solana_program::instruction::CompiledInstruction};
+
+pub fn is_simple_vote_transaction<'a>(
+    signatures_count: usize,
+    instructions_count: usize,
+    is_legacy_message: bool,
+    mut instructions: impl Iterator<Item = (&'a Pubkey, &'a CompiledInstruction)>,
+) -> bool {
+    if signatures_count < 3 && instructions_count == 1 && is_legacy_message {
+        instructions.next().map(|(program_id, _ix)| program_id) == Some(&crate::vote::program::id())
+    } else {
+        false
+    }
+}
```

Comment:
I think `instructions_count` is redundant w/ the instructions iterator:


```rust
    if signatures_count < 3 && is_legacy_message {
        let first_ix = instructions.next();
        let next_ix = instructions.next();
        if next_is.is_none() {
           first_ix.map(|(program_id, _ix)| program_id) == Some(&crate::vote::program::id())
        } else {
			false
        }
    } else {
        false
    }
```

alternatively, using `Option::xor`:
```rust
	signatures_count < 3 && is_legacy_message &&
	    instructions.next()
	        .xor(instructions.next())
	        .map(|(program_id, _ix)| program_id == &crate::vote::program::id())
	        .unwrap_or_false()
```

`Option::xor` and how iterators work let's us guarantee that the the first instruction exclusive!

Reply:
🆒 the usage of `option::xor` over iterator! 

But just for the sake of debate, `iterator` itself does not contains `size` information, we could ofc _try_ to extract that info as demo above, but not necessarily reliable. For example, the iter parametyer could be the last iter of !vec(ix; 3)...

With that said, this function does look ugly as it is rn.

Reply:
> For example, the iter parametyer could be the last iter of !vec(ix; 3)...

Yeah, possible but less likely imo. My main point was that we've got 2 args which give us very related info. Easier to screw up when 2 args need consistency

Reply:
Perhaps it can also be argued that although these 2 arg may derived from same vector, but they don't have to be related, especially from this function perspective - it wants to have 1) the size of the vector, 2) iterator of the vector. If can have an iter that is `head` of vector, then that one arg can provide both info.